src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -214,7 +219,8 @@ private static void AddLinuxContainerServices(this IServiceCollection services)
             services.AddSingleton<IHostedService>(s =>
             {
                 var environment = s.GetService<IEnvironment>();
-                if (environment.IsLinuxConsumption())
+                //todo: Replace with legion specific service
```

Comment:
this comment may be misleading? When legion logic is added, you'll be adding a new if branch here, not replacing all this logic right?

Also - is there a tracking issue for doing the follow up work tracked by these TODOs? It'd be good to have that issue and reference it in all these TODOs so it's all linked up and can be cleaned up in one go.

Reply:
Yes, will start adding some tasks here. currently they are tracked in dev ops.